boards: renesas: Add support Renesas ek_ra8m2 board

khoa-nguyen-18 opened the pull request:
- Add support Renesas RA8M2 SoC
- Add support Renesas `ek_ra8m2_r7ka8m2jflcac_cm85` board
- Add support tests and samples app to verify `ek_ra8m2_r7ka8m2jflcac_cm85` peripherals

khoa-nguyen-18 committed:
soc: renesas: ra: Add support Renesas ra8m2 SoC

Add support Renesas ra8m2 SoC

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>
Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoa-nguyen-18 committed:
boards: renesas: Add support Renesas ek_ra8m2 board

Add support Renesas ek_ra8m2 board

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>
Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoa-nguyen-18 committed:
tests: drivers: i2c: Add support for i2c_api on ek_ra8m2

Add support for test app `i2c_api` on Renesas ek_ra8m2

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>
Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoa-nguyen-18 committed:
tests: drivers: comparator: Add support gpio_loopback on ek_ra8m2

Add support for test app `gpio_loopback` on Renesas ek_ra8m2

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

khoa-nguyen-18 committed:
tests: drivers: pwm: Add support for pwm tests on ek_ra8m2

Add support for test apps on Renesas ek_ra8m2:
- tests/drivers/pwm_api
- tests/drivers/pwm_loopback

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>
Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoa-nguyen-18 committed:
tests: drivers: spi: Add support for spi_loopback on ek_ra8m2

Add support for test app `spi_loopback` on Renesas ek_ra8m2

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>
Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoa-nguyen-18 committed:
tests: drivers: uart: Add support for uart_async_api on ek_ra8m2

Add support for test app `uart_async_api` on Renesas ek_ra8m2

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>
Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoa-nguyen-18 committed:
tests: subsys: pm: Add support for power_mgmt_soc on ek_ra8m2

Add support for test app `power_mgmt_soc` on Renesas ek_ra8m2

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>
Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoa-nguyen-18 committed:
samples: drivers: counter: Add support for alarm on ek_ra8m2

Add support for sample app `alarm` on Renesas ek_ra8m2

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

khoa-nguyen-18 committed:
samples: boards: renesas: Add support for comparator on ek_ra8m2

Add support for sample app `renesas/comparator` on ek_ra8m2

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

khoatranyj committed:
tests: drivers: counter: Add support test on Renesas ek_ra8m2

Add support for test app on Renesas ek_ra8m2:
- tests/drivers/counter/counter_basic_api

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: i2s: Add tests support on Renesas ek_ra8m2

Add support for i2s test app on Renesas RA ek_ra8m2:
- tests/drivers/i2s/i2s_api
- tests/drivers/i2s/i2s_speed

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
samples: drivers: i2s: Add tests support for i2s ssie on ek_ra8m2

Add support for i2s samples app on Renesas RA ek_ra8m2:
- samples/drivers/i2s/output

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: dma: Add test support dma driver on ek_ra8m2

This commit adds testing support for dma driver on ek_ra8m2:
- tests/drivers/dma/chan_blen_transfer
- tests/drivers/dma/loop_transfer

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
dts: arm: renesas: ra: Add support Renesas r7ka8m2jflcac SoC

Add support Renesas r7ka8m2jflcac SoC

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>
Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: sdhc: Add tests support for Renesas ek_ra8m2 board

Add tests support for Renesas ek_ra8m2 boards

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: subsys: sd: Add tests support for Renesas ek_ra8m2 board

Add tests support for Renesas ek_ra8m2 boards:
- tests/subsys/sd/sdmmc

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: driver: disk: Add tests support for Renesas ek_ra8m2 board

Add tests support for Renesas ek_ra8m2 boards:
- tests/drivers/disk/disk_access
- tests/drivers/disk/disk_performance

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: subsys: fs: Add tests support for Renesas ek_ra8m2 board

Add tests support for Renesas ek_ra8m2 boards:
- tests/subsys/fs/ext2
- tests/subsys/fs/fat_fs_api

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
samples: subsys: fs: Add tests support for Renesas ek_ra8m2 board

Add tests support for Renesas ek_ra8m2 boards:
- samples/subsys/fs

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

quytranpzz commented on dts/arm/renesas/ra/ra8/r7ka8m2xf.dtsi:
```diff
@@ -0,0 +1,803 @@
+
+			cpuclk0: cpuclk0 {
+				compatible = "renesas,ra-cgc-pclk";
+				clock-frequency = <1000000000>;
```
```suggestion
				clock-frequency = <DT_FREQ_M(1000)>;
```

thaoluonguw commented on west.yml:
```diff
@@ -226,7 +226,7 @@ manifest:
     - name: hal_renesas
       path: modules/hal/renesas
-      revision: 5ab2c84af5cbdbc2b1f0a41e08b8f311bd4eafa8
+      revision: pull/162/head
```
hal_renesas which support for RA8M2 was merge. Please update revision.

thaoluonguw commented on soc/renesas/ra/ra8m2/Kconfig:
```diff
@@ -0,0 +1,27 @@
+	select CPU_CORTEX_M_HAS_DWT
+	select ARMV8_M_DSP
+	select ARMV8_1_M_MVEI
+	select ARMV8_1_M_MVEF
```
These configs depend on ARMV8_1_M_MAINLINE & ARMV8_M_DSP are available, skip this selection here.

khoa-nguyen-18 committed:
samples: modules: lgvl: Update code mram for Renesas ek_ra8p1

Since the sample build is oversized for the RA8P1 CM85,
I’ve realigned the code MRAM size for both cores (CM85 and CM33)
to allow the CM85 build to run the sample successfully

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

thaoluonguw reviewed (dismissed):
LGTM now. Thank you
